### ISSUE TITLE
Add EventList plots and tests

### ISSUE THREAD

**Issue by cdeil** [gammapy/gammapy#409]
The [gammapy.data.EventList](https://gammapy.readthedocs.org/en/latest/api/gammapy.data.EventList.html) class for IACT event lists needs some love.
- Add tests (see current [coverage](https://coveralls.io/builds/4607932/source?filename=gammapy%2Fdata%2Fevent_list.py)).
- Add a few standard plots as quick ways to see if the event data looks OK (implement the [peek](https://gammapy.readthedocs.org/en/latest/_modules/gammapy/data/event_list.html#EventList.peek) method and the plots mentioned there)).
- The most important plot IMO is the offset vs. energy one mentioned in #297.
- I saw this [blog post on timemaps](https://districtdatalabs.silvrback.com/time-maps-visualizing-discrete-events-across-many-timescales) and think it could be nice to make a time map plot also for event lists (see [prototype](http://nbviewer.ipython.org/gist/cdeil/f0b5c585bb78d7096a53)).
- Add a docs page `docs/data/events.html` that gives a 1 min tutorial on how to work with event lists, showing off these plots.

This could easily be split in several small PRs (one for adding tests, one for each new plot) or be one larger PR. @JonathanDHarris - Would you be interested to work on this?

If there's someone interested to work on this, we can discuss specifics (i.e. what the tests should do exactly or the plots should show exactly). The plots could be either matplotlib (like we did so far). Or, if someone wants to try it out, we could start using Bokeh to make interactive plots. Bokeh even allows linking tables to plots in the web browser with very little code (see example: https://vine.co/v/OvUJ9iIP5u7).

For now this event list can be used for tests / examples:
https://github.com/gammapy/gammapy-extra/raw/master/datasets/hess-crab4/hess_events_simulated_023523.fits

Hopefully a better, real HESS event list will be released soon and then we can switch to that one.

**Comment by JonathanDHarris** [comment]
@cdeil Yep I'm interested.  I'll probably do several small prs.

As it's been a while, could you link me again to the idiots guide to your git flow?

**Comment by cdeil** [comment]
I haven't written up any dev or git info for Gammapy specifically.

There's a ton of pages of docs though for Astropy and it's the same for Gammapy.

Does this help?
http://astropy.readthedocs.org/en/stable/development/workflow/development_workflow.html#astropy-git

If not, feel free to ask any questions on Github or the mailing list!
I've also been thinking about setting up https://slack.com/ for Gammapy dev, but I wasn't sure if it helps or distracts to have a separate page from GH issue / pull requests to chat / discuss things.
@JonathanDHarris - Would you be interested in trying out Slack or should we stick to GH / email?

**Comment by JonathanDHarris** [comment]
Thanks @cdeil that link was what I was after.

I'm happy to give slack a try

**Comment by cdeil** [comment]
Does this work?
https://gammapy.slack.com/

Is it publicly visible?

**Comment by JonathanDHarris** [comment]
Slack doesn't appear to be public.

'To start using Slack, you'll need to be invited by your team administrator.'

Is there a guide to installing gammapy-extra?

**Comment by JonathanDHarris** [comment]
So, when I code at work we break down tests into very, very specific methods.

E.g. we might have

test_EventList_returns_correct_ra_and_dec

Do you want me to do that here, or do you prefer to stick with a broad test_EventList ?

**Comment by cdeil** [comment]
Yes, it's better to split tests in small functions or methods that each test one thing.
On the other hand, I don't think we have to try to implement unit tests for everything, i.e. it's good enough to have high-level tests that show that it basically works.
But those should be < 10 lines long, if more code is needed it should be split out into helper functions.

At the moment there are two tests for EventList here that look OK to me:
https://github.com/gammapy/gammapy/blob/master/gammapy/data/tests/test_event_list.py

So bottom line: yes, feel free to add many small tests. But if someone prefers slightly longer tests that's OK as well.

**Comment by cdeil** [comment]
Closing this old issue. No concrete action items here. We have other issues do discuss event lists still open.
